tests/test_webhooks.py

```
@@ -0,0 +1,331 @@
+        timestamp=timestamp,
+    )
+
+    # No X-GengoWatcher-Timestamp header - should reject even with valid signature
+    from gengowatcher.webhooks import WebhookSignatureError
+    with pytest.raises(WebhookSignatureError):
+        verify_webhook_signature(
+            raw_body=raw_body,
+            headers={
+                "X-GengoWatcher-Signature": signature,
+            },
+            secret="secret",
```

Ruff (0.15.20)

[error] Possible hardcoded password assigned to argument: "secret"

(S106)